connaisseur/config.py

```diff
@@ -27,55 +31,72 @@ def __init__(self):
         with open(self.__PATH, "r") as configfile:
             config_content = yaml.safe_load(configfile)
 
-        if not config_content:
+        with open(self.__SECRETS_PATH, "r") as secrets_configfile:
+            secrets_config_content = yaml.safe_load(secrets_configfile)
+
+        config = self.__merge_configs(config_content, secrets_config_content)
+
+        self.__validate(config)
+
+        self.validators = [Validator(**validator) for validator in config]
+
+    def __merge_configs(self, config: dict, secrets_config: dict):
+        if not config:
             msg = "Error loading connaisseur config file."
             raise NotFoundException(message=msg)
 
-        self.__validate(config_content)
-
-        self.notaries = [Notary(**notary) for notary in config_content]
+        for validator in config:
+            validator.update(secrets_config.get(validator.get("name"), {}))
+            auth_path = f'{self.__EXTERNAL_PATH}{validator["name"]}/auth.yaml'
```

Comment:
The validation hasn't run yet. It may be safe -- but are we sure this is safe?

Reply:
I feel like we are doing all we can. Ppl can't break out of the directory through our `safe_path_func` and whatever funny business they put into the `secret_config` or `auth_file` will be validated against the schema in the end. They can overwrite existing values however they want, it still needs to be schema conform. I don't think doing the validation for each file on its own would change much here.

Reply:
I suggest to add a comment here that the schema validation hasn't run, and that we're only using a name parameter that is confined with `safe_path_func`.

Just to make sure that when this is later changed, no one will think that anything in `validator` is safe to use just because we're using `validator["name"]`. That is, let's be explicit about the "exceptional circumstance" that allows us to do this. (Exceptions confirm the rule! ;-) )

Reply:
ah damn, i just realized, since validation hasn't run through we can't be sure a `name` field actually exists yet. so I better use `validator.get("name")` instead.